creator-node/src/routes/tracks.js

```diff
@@ -37,12 +40,90 @@ const RehydrateIpfsQueue = require('../RehydrateIpfsQueue')
 const { FileProcessingQueue } = require('../FileProcessingQueue')
 const DBManager = require('../dbManager')
 const { generateListenTimestampAndSignature } = require('../apiSigning.js')
+const DiskManager = require('../diskManager')
 
+const server = new tus.Server()
 const readFile = promisify(fs.readFile)
+const tmpTrackArtifactsPath = DiskManager.getTmpTrackUploadArtifactsPath()
+const resumableUploadRoute = path.join(tmpTrackArtifactsPath, '*', '*')
+
+// Structure: <uuid>.<file_extension>. e.g. 1234-1234-1234-1234.mp3
+const getFileName = (req) => {
+  return req.headers.randomfilename + getFileExtension(req.headers.filename)
+}
 
 const SaveFileToIPFSConcurrencyLimit = 10
 
 module.exports = function (app) {
+  app.post('/track_content_upload', authMiddleware, ensurePrimaryMiddleware, ensureStorageMiddleware, syncLockMiddleware, checkFileMiddleware, async function (req, res, next) {
+    // Use the tus-node-server package to handle track upload
+
+    // Save file under randomly named folders to avoid collisions
+    const fileDir = path.join(tmpTrackArtifactsPath, req.headers.randomfilename)
```

Comment:
lets pull all the header values out here and check for existence

Reply:
I can add that to the middleware!